components/buttons/NewConversationButton.tsx

```diff
@@ -65,7 +68,18 @@ export default function NewConversationButton({
       setErrorMessage('');
       setError(false);
 
-      await startNewConversation(userId, donorId, item_id);
+      const conversationId = await startNewConversation(
+        userId,
+        donorId,
+        item_id
+      );
+
+      if (conversationId) {
+        dispatch({
+          type: 'SET_CURRENT_CONVERSATION_ID',
+          payload: conversationId,
+        });
```

Comment:
Fantastic!!🔥 Thank you @camelPhonso !! Your advice is ingenious and simple. I spent a lot of time to get into the code and didn't come up with anything better to add a new variable🫣 A little ashamed of myself😳🥴. I checked your tips and they need to be improved a bit: 
   We may not have **allConversations** unless the user opens the **ConversationWrapper** where all the conversations are fetched... so we'll have to do **fetchedConversations** somewhere here or ... But there's a better idea, but I haven't tested it yet. I suggest to change the function **startNewConversation** so that it returns not the **conversationId**, but the **conversation object**, in order to immediately send **'SET_CURRENT_CONVERSATION'** and it will not be necessary to extract **allConversations**. 
   ☝️P.S: I previously supplemented the **startNewConversation** function with the expectation that we would use the **conversation id**, I didn't realize then that we were using the **conversation object** and not the id in the code. As long as I don't see that we use the conversation id directly somewhere, then such a change will be appropriate.

Reply:
> I spent a lot of time to get into the code and didn't come up with anything better to add a new variable🫣 A little ashamed of myself😳🥴.

Not at all! The reducer stuff is a bit convoluted at first and I always have to spend a few minutes fishing around the code base to remember how it's being set up myself 🤣 - I expect everyone will need a little while to get used (though it makes everything else super simple once you're using it)

I personally tend to have the types files open to the side when I'm working with the context just to make sure I remind myself what information is stored in these states 💡 

> We may not have allConversations unless the user opens the ConversationWrapper where all the conversations are fetched...

The context provider runs a check whenever you try to import `useConversationContext` and throws an error if you are outside of reach for the context, so it should be that if you are calling the `dispatch` we should be able to get `allConversations`. Nothing like trying it out and see if everything breaks though 😅 

> But there's a better idea, but I haven't tested it yet. I suggest to change the function startNewConversation so that it returns not the conversationId, but the conversation object, in order to immediately send 'SET_CURRENT_CONVERSATION' and it will not be necessary to extract allConversations

Yep, if we need to make changes I would have thought something like this as well ☝️ 

Give it a test to try and get `allConversations` first and then if there's any issues run with your plan 👍

Reply:
I tested this by adding:
const { conversationState: {
 allConversations
 }, dispatch } = useConversationContext();

but allConversations is empty until I open "/conversations". Therefore, when the user opens the application, and wants to open and start a conversation, allConversations will be empty, because it has not been downloaded yet. So it didn't work for me until I did it
...
const fetchedConversations = await selectedUserConversations(userId);
 dispatch({
 type: 'SET_ALL_CONVERSATIONS',
 payload: fetchedConversations,
 });
...
What do you think about it?

Reply:
Please check me if I am thinking correctly:
1. function startNewConversation creates a new conversation or returns the conversationId of an existing conversation.
2. allConversations will be empty until filled with the appropriate action('SET_ALL_CONVERSATIONS'), which happens when the ConversationWrapper is rendered when "/conversations" is opened.
3. if we want to get the corresponding conversation, we need to refer to the database after we create a new conversation using startNewConversation, so we need to update the state of allConversations...
4. regarding setting the current conversation: when opening "/conversations" it is intended that the current conversation will always be the most recent conversation from the list of conversations (allConversations[0]). This does not take into account that we can open an already existing conversation from the item page. This needs to be fixed.

Reply:
> but allConversations is empty until I open "/conversations". 

Ok yeah, see we all get confused 😅  You're absolutely right.

Alright, I've spent a few minutes looking at this and I think you've hit the solution already:

1. Return the full conversation from `startNewConversation`
2. Dispatch `SET_SHOW_CONVERSATIONS_LIST` and `SET_CURRENT_CONVERSATION` from `NewConversationButton.tsx`
3. Check if `CURRENT_CONVERSATION` is set when we open`CurrentConversation.tsx` and if not, then set it to `allConversations[0]`

When we start a new conversation it gets placed as the latest in the database and is retrieved at index 0 when we first load the `ConversationsWrapper` but if we are navigating to a conversation that was already set before then we will have set the value of `currentConversation` already and just open that one.

Sorry for introducing a bit of confusion xD I think you're on the right track 👍

Reply:
Thank you very much @camelPhonso. I really appreciate your time. You do not confuse me at all, on the contrary, you help me think correctly:-) Your advice is as always very appropriate, I will try to implement it all.